pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -108,31 +108,32 @@ type EFIConfiguration struct {
 }
 
 type ConverterContext struct {
-	Architecture          string
-	AllowEmulation        bool
-	Secrets               map[string]*k8sv1.Secret
-	VirtualMachine        *v1.VirtualMachineInstance
-	CPUSet                []int
-	IsBlockPVC            map[string]bool
-	IsBlockDV             map[string]bool
-	HotplugVolumes        map[string]v1.VolumeStatus
-	PermanentVolumes      map[string]v1.VolumeStatus
-	DisksInfo             map[string]*cmdv1.DiskInfo
-	SMBios                *cmdv1.SMBios
-	SRIOVDevices          []api.HostDevice
-	GenericHostDevices    []api.HostDevice
-	GPUHostDevices        []api.HostDevice
-	EFIConfiguration      *EFIConfiguration
-	MemBalloonStatsPeriod uint
-	UseVirtioTransitional bool
-	EphemeraldiskCreator  ephemeraldisk.EphemeralDiskCreatorInterface
-	VolumesDiscardIgnore  []string
-	Topology              *cmdv1.Topology
-	ExpandDisksEnabled    bool
-	UseLaunchSecurity     bool
-	FreePageReporting     bool
-	BochsForEFIGuests     bool
-	SerialConsoleLog      bool
+	Architecture                    string
+	AllowEmulation                  bool
+	Secrets                         map[string]*k8sv1.Secret
+	VirtualMachine                  *v1.VirtualMachineInstance
+	CPUSet                          []int
+	IsBlockPVC                      map[string]bool
+	IsBlockDV                       map[string]bool
+	HotplugVolumes                  map[string]v1.VolumeStatus
+	PermanentVolumes                map[string]v1.VolumeStatus
+	DisksInfo                       map[string]*cmdv1.DiskInfo
+	SMBios                          *cmdv1.SMBios
+	SRIOVDevices                    []api.HostDevice
+	GenericHostDevices              []api.HostDevice
+	GPUHostDevices                  []api.HostDevice
+	EFIConfiguration                *EFIConfiguration
+	MemBalloonStatsPeriod           uint
+	UseVirtioTransitional           bool
+	EphemeraldiskCreator            ephemeraldisk.EphemeralDiskCreatorInterface
+	VolumesDiscardIgnore            []string
+	Topology                        *cmdv1.Topology
+	ExpandDisksEnabled              bool
+	UseLaunchSecurity               bool
+	FreePageReporting               bool
+	BochsForEFIGuests               bool
+	SerialConsoleLog                bool
+	DomainAttachmentByInterfaceName map[string]string
```

Comment:
I really do not like this context struct. Everyone is touching it when something new is added.
We need to somehow decouple the network part from all this mess in the future.

Reply:
+1
`CreateDomainInterface` use this struct only for the `UseLaunchSecurity` boolean, it can easily be decoupled by passing it directly, same goes for `DomainAttachmentByInterfaceName`.
We can do it in a follow up PR, if you dont what to do it in this PR.